pkg/cmd/pulumi/destroy.go

```diff
@@ -203,6 +214,9 @@ func newDestroyCmd() *cobra.Command {
 	cmd.PersistentFlags().BoolVar(
 		&suppressPermaLink, "suppress-permalink", false,
 		"Suppress display of the state permalink")
+	cmd.PersistentFlags().BoolVar(
+		&optInNonSaasPermaLink, "non-saas-permalink-enabled", false,
```

Comment:
Two things:
1. We don't use the team "non-saas" anywhere else.  We use "self-managed" in the backends docs, and "filestate" in the implementation.  I'm not sure exactly what's right here - but we should likely base around one of these names instead of introducing "non-saas"
2. We already have `suppress-permalink`.  It feels like this new option should be `suppress-permalink=false`.  We are basically making the defalult for `suppress-permalink` true for the filestate backend with these changes. Not sure exactly how best to represent that with Cobra.

Reply:
so `suppress-permalink` still works - as we allow suppressing even for hosted backend. I can't make it work with both. The default for this is `false`. Changing this to true gives us chicken and egg based on the hosted backend. I can't figure out a way of being able to use the existing flag without breaking those who use the hosted backend so I was trying to create an explicit flag to make it opt in for self-managed backends

(FWIW, I like selfManagedBackends rather than non-SaaS so thanks for that(

Reply:
Ah, the naming is a good point. We're starting to circle around a final set of names for the various backends here: https://docs.google.com/document/d/10cLvyua-xJzVEuZjY0InfLPYeF2RQtG_BLdqUun35D4/edit#heading=h.q32ufnmp2er2

Lemme see if I can draw us to a close there so we can start using those new names.

Reply:
Let's stick with `selfManagedBackends` for now - we should have a new name sorted here in the next few weeks, but I don't want to block this PR in the meantime.